pages/index.vue

```diff
@@ -1,67 +1,21 @@
 <script lang="ts" setup>
-import { LanguageEnum } from "@bcc-code/bmm-sdk-fetch";
-import type { DiscoverGetRequest, TrackModel } from "@bcc-code/bmm-sdk-fetch";
-import { breakpointsTailwind, useBreakpoints } from "@vueuse/core";
-import type { IDiscoverableGroup } from "~/composables/discover";
-
-const { setQueue } = useNuxtApp().$mediaPlayer;
-const userData = useNuxtApp().$userData;
+import type { DiscoverGetRequest, LanguageEnum } from "@bcc-code/bmm-sdk-fetch";
 
 const { t, locale } = useI18n();
+const userData = useNuxtApp().$userData;
 toolbarTitleStore().setReactiveToolbarTitle(() => t("nav.home"));
 
-const discoverGroups = ref<IDiscoverableGroup[] | null>(null);
-const loading = ref(true);
-
-let stopHandles: (() => void)[] = [];
-watch(
-  locale,
-  () => {
-    const parameters: DiscoverGetRequest = {
-      // Compatibility is ensured in i18n.config.ts file
-      lang: locale.value as LanguageEnum,
-    };
-    if (userData.age) parameters.age = userData.age;
-    const { data, pending, stopHandler } = useDiscover(parameters);
-    stopHandles.forEach((el) => el());
-
-    stopHandles = [
-      stopHandler,
-      watch(
-        data,
-        (d) => {
-          discoverGroups.value = d;
-        },
-        { immediate: true },
-      ),
-      watch(
-        pending,
-        (p) => {
-          loading.value = p;
-        },
-        { immediate: true },
-      ),
-    ];
-  },
-  { immediate: true },
-);
-
-const playItem = (item: TrackModel, group: IDiscoverableGroup) => {
-  const items = group.items.filter((c): c is TrackModel => c.type === "track");
-  setQueue(
-    items,
-    items.findIndex((track) => track.id === item.id),
-  );
+const parameters: DiscoverGetRequest = {
+  // Compatibility is ensured in i18n.config.ts file
+  lang: locale.value as LanguageEnum,
 };
-
-const breakpoints = useBreakpoints(breakpointsTailwind);
-
-const isSmallScreen = breakpoints.smallerOrEqual("lg");
+if (userData.age) parameters.age = userData.age;
+const { data: models, pending } = useDiscover(parameters);
```

Comment:
Having this not wrapped in a `watch` will not load this again when the language is changed. In my opinion, we should keep the behavior, otherwise it looks like the headlines of the home screen do not adjust the language when changed, but the rest does.

Reply:
true.

I actually also need to add that it's translated on the server. Those endpoints still use an older concept where the server only sends the translation id.

Reply:
I added UiLanguage as Http header so that the Api knows the language for every request.
So sending the language as QueryParameter is no longer needed (e.g. see Discover request).